kyuubi-util-scala/src/main/scala/org/apache/kyuubi/util/MathUtils.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.util
+
+import scala.util.Try
+
+object MathUtils {
+
+  /**
+   * Check if the given object is a positive number
+   * @param o the object to be checked
+   * @return true if the object is a positive number
+   */
+  def isPositiveNumber(o: Any): Boolean = o match {
+    case i: Int => i > 0
+    case l: Long => l > 0
+    case f: Float => f > 0
+    case d: Double => d > 0
+    case _ => Try { o.toString.toDouble > 0 }.isSuccess
+  }
+
+  /**
+   * Check if the given object is a non-negative number
+   *
+   * @param o the object to be checked
+   * @return true if the object is a non-negative number
+   */
+  def isNonNegativeNumber(o: Any): Boolean = o match {
+    case intValue: Int => intValue >= 0
+    case longValue: Long => longValue >= 0L
+    case floatValue: Float => floatValue >= 0.0f
+    case doubleValue: Double => doubleValue >= 0.0d
+    case byteValue: Byte => byteValue >= 0
+    case shortValue: Short => shortValue >= 0
+    case charValue: Char => charValue >= 0
+    case _ => Try { o.toString.toDouble >= 0.0d }.isSuccess
```

Comment:
the logic is incorrect here, it only guards the type cast, ignores the expression result

Reply:
BTW, should we narrow the `o`'s types?

Reply:
> the logic is incorrect here, it only guards the type cast, ignores the expression result

Yes, noticed. Fixed and added proper unit test for this.

Reply:
> BTW, should we narrow the `o`'s types?

In what way or what's the ranger of types?

Reply:
Are Int, Long, Float, Double, String sufficient?

Reply:
Primitive data types have been covered, meanwhile other types are converted to string and then to a double for comparison. Should be good enough for all cases.

Reply:
I mean narrow the range of data types

Reply:
Adopted your suggestion. And it will throw an IllegalArgumentException for unsupported types.